src/main/java/eu/dissco/backend/repository/DigitalMediaObjectRepository.java

```diff
@@ -1,103 +1,225 @@
+        .offset(offset).limit(pageSize).fetch(this::mapToMultiMediaObject);
+  }
+
+  public List<JsonApiData> getDigitalMediaObjectJsonResponse(int pageNumber, int pageSize) {
+    var offset = 0;
+    if (pageNumber > 1) {
+      offset = offset + (pageSize * (pageNumber - 1));
+    }
+
+    Field<Integer> maxVersion = DSL.max(NEW_DIGITAL_SPECIMEN.VERSION).as("maxVersion");
+
+    var specimenRecentVersions = context.select(NEW_DIGITAL_SPECIMEN.ID, maxVersion)
+        .from(NEW_DIGITAL_SPECIMEN).groupBy(NEW_DIGITAL_SPECIMEN.ID).asTable();
+
+    return context.select(NEW_DIGITAL_SPECIMEN.SPECIMEN_NAME, NEW_DIGITAL_SPECIMEN.VERSION,
+            NEW_DIGITAL_SPECIMEN.ID, NEW_DIGITAL_MEDIA_OBJECT.asterisk()).from(NEW_DIGITAL_SPECIMEN)
+        .join(specimenRecentVersions)
+        .on(NEW_DIGITAL_SPECIMEN.ID.eq(specimenRecentVersions.field(NEW_DIGITAL_SPECIMEN.ID)))
+        .and(NEW_DIGITAL_SPECIMEN.VERSION.eq(specimenRecentVersions.field(maxVersion)))
+        .join(NEW_DIGITAL_MEDIA_OBJECT)
+        .on(NEW_DIGITAL_SPECIMEN.ID.eq(NEW_DIGITAL_MEDIA_OBJECT.DIGITAL_SPECIMEN_ID)).offset(offset)
+        .limit(pageSize).fetch(this::mapToJsonApiData);
+  }
+
+  public int getMediaObjectCount(int pageSize) {
+    int totalRecords = context.selectCount().from(NEW_DIGITAL_MEDIA_OBJECT).fetchOne(0, int.class);
+    return totalRecords / pageSize + ((totalRecords % pageSize == 0) ? 0 : 1);
   }
 
   public List<String> getDigitalMediaIdsForSpecimen(String id) {
-    return context.select(NEW_DIGITAL_MEDIA_OBJECT.ID)
-        .distinctOn(NEW_DIGITAL_MEDIA_OBJECT.ID)
-        .from(NEW_DIGITAL_MEDIA_OBJECT)
-        .where(NEW_DIGITAL_MEDIA_OBJECT.DIGITAL_SPECIMEN_ID.eq(id))
+    return context.select(NEW_DIGITAL_MEDIA_OBJECT.ID).distinctOn(NEW_DIGITAL_MEDIA_OBJECT.ID)
+        .from(NEW_DIGITAL_MEDIA_OBJECT).where(NEW_DIGITAL_MEDIA_OBJECT.DIGITAL_SPECIMEN_ID.eq(id))
         .orderBy(NEW_DIGITAL_MEDIA_OBJECT.ID, NEW_DIGITAL_MEDIA_OBJECT.VERSION.desc())
         .fetch(Record1::value1);
   }
+
+  private DigitalMediaObject mapToMultiMediaObject(Record dbRecord) {
+    try {
+      return new DigitalMediaObject(dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.ID),
+          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.VERSION),
+          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.CREATED),
+          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.TYPE),
+          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.DIGITAL_SPECIMEN_ID),
+          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.MEDIA_URL),
+          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.FORMAT),
+          dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.SOURCE_SYSTEM_ID),
+          mapper.readTree(dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.DATA).data()),
+          mapper.readTree(dbRecord.get(NEW_DIGITAL_MEDIA_OBJECT.ORIGINAL_DATA).data()));
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private JsonApiData mapAnnotationToJsonApiData(Record dbRecord) {
```

Comment:
If you let annotations handle annotation than you would not have to use this method but can just use the functionality that is already there. Now it is duplicated. If this needs to stay here I would say we move this method to a shared method in a parent class